tools/ack-discover/ecrpublic.py

```diff
@@ -0,0 +1,136 @@
+        RoleArn=ECR_PUBLIC_READER_ROLE_ARN,
+        RoleSessionName=role_session_name,
+        DurationSeconds=60*60,  # 1 hour...
+    )
+    creds = resp['Credentials']
+    return AWSCredentials(
+        access_key_id=creds['AccessKeyId'],
+        secret_access_key=creds['SecretAccessKey'],
+        session_token=creds['SessionToken'],
+    )
+
+
+def get_client(writer):
+    """Returns the ECR Public client object after assuming the ECR Public
+    Reader role.
+    """
+    aws_creds = _assume_reader_role(writer)
+
+    # NOTE(jaypipes): ECR Public requires "us-east-1" when calling
+    # DescribeRepositories...
+    return boto3.client(
+        "ecr-public",
+        region_name="us-east-1",
+        aws_access_key_id=aws_creds.access_key_id,
+        aws_secret_access_key=aws_creds.secret_access_key,
+        aws_session_token=aws_creds.session_token,
+    )
+
+
+@dataclasses.dataclass
+class Repository:
+    registry_id: str
+    name: str
+    uri: str
+    created_on: str
+
+
+def get_repository(writer, ep_client, repo_url):
+    """Returns an Repository describing the supplied repository
+    URL or None if no such repository exists.
+    """
+    writer.debug("[get_public_repository] ", repo_url)
+    parts = repo_url.split("/")
+    registry_id = ECR_PUBLIC_AWS_ACCOUNT_ID
+    repo_name = parts[2]
+    try:
+        repos = ep_client.describe_repositories(
+            registryId=registry_id,
+            repositoryNames=[repo_name],
+        )
+        repo_data = repos["repositories"][0]
+        return Repository(
+            registry_id=registry_id,
+            name=repo_name,
+            uri=repo_data["repositoryUri"],
+            created_on=repo_data["createdAt"],
+        )
+    except ep_client.exceptions.RepositoryNotFoundException:
+        return None
+
+
+def get_repository_latest_tag(ep_client, repo):
+    """Returns the image tag of the latest pushed image in a supplied repo."""
+    most_recent = None
+    latest_tag = None
+    try:
+        images = ep_client.describe_images(
+            registryId=repo.registry_id,
+            repositoryName=repo.name,
+        )
+        for image in images["imageDetails"]:
+            pushed_at = image["imagePushedAt"]
+            if most_recent is None or pushed_at > most_recent:
+                most_recent = pushed_at
+                latest_tag = image["imageTags"][0]
+        return latest_tag
+    except ep_client.exceptions.RepositoryNotFoundException:
+        return latest_tag
+
+
+def chart_has_stable_tag(ep_client, chart_repo):
+    """Returns True if the supplied chart repository has a chart with a stable
+    tag, False otherwise.
+    """
+    try:
+        images = ep_client.describe_images(
+            registryId=chart_repo.registry_id,
+            repositoryName=chart_repo.name,
+        )
+        for image in images["imageDetails"]:
+            if "imageTags" not in image:
+                continue
+            tag = image["imageTags"][0]
```

Comment:
Stable won't always be the latest tag? [`describe_images`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecr-public.html#ECRPublic.Client.describe_images) seems to accept a filter field for this? 
```python
imageIds=[{'imageTag': 'stable'}]
```

Reply:
Yeah, was mostly just a copy/paste from the above `get_repository_latest_tag` function :) I can optimize this to use the imageTags filter.